### ISSUE TITLE
Add a faster version for shortcircuit calculation for single bus

### ISSUE THREAD

**Issue by wangzhenassd** [e2nIEE/pandapower#1037]
Please checkout branch single_sc

It should only work for voltage_source at the moment.

Address also issue #1033

**Comment by wangzhenassd** [comment]
A simple example
```python
from copy import deepcopy
import pandapower as pp
from pandapower.shortcircuit import calc_sc, calc_single_sc
import pandapower.networks as nw

net = nw.create_cigre_network_mv()
net.ext_grid['s_sc_max_mva'] = 1000
net.ext_grid['rx_max'] = 0.1
net.sgen = net.sgen.iloc[0:0, :]

calc_single_sc(net,5)        
calc_single_sc(net_no_inv, 5, with_y_inv=False)
    
print(net.res_bus)
print(net.res_bus_sc)
```

**Comment by wangzhenassd** [comment]
It also seems to work with larger system:
```
    net = nw.case1354pegase()    
    net.ext_grid['s_sc_max_mva'] = 1000
    net.ext_grid['rx_max'] = 0.1
    net.gen["vn_kv"] = net.bus.loc[net.gen.bus.values, "vn_kv"].values
    net.gen["rdss_pu"] = 0.01
    net.gen["xdss_pu"] = 0.1
    net.gen["cos_phi"] = 0.8
    net.gen["sn_mva"] = net.gen.p_mw.values
    
    net.sgen['k'] = 1.
    net.sgen = net.sgen.iloc[0:0, :]

    net_no_inv = deepcopy(net)

    calc_single_sc(net,254)        
    calc_single_sc(net_no_inv, 254, with_y_inv=False)

    print((net.res_bus_sc - net_no_inv.res_bus_sc).max())
```

**Comment by bergkvist** [comment]
It seems like the `calc_single_sc` is not yet implemented for 1ph-short circuits. This is also something that would be nice to have.

#### `calc_sc`:
https://github.com/e2nIEE/pandapower/blob/c9aaecba98c6a0c812452a82fb7064b82bde1c70/pandapower/shortcircuit/calc_sc.py#L128-L133

#### `calc_single_sc`:
https://github.com/e2nIEE/pandapower/blob/c9aaecba98c6a0c812452a82fb7064b82bde1c70/pandapower/shortcircuit/calc_sc.py#L198-L201

**Comment by wangzhenassd** [comment]
#1082 should fulfil the first phase of the improved short circuit calculation functionality

TODO:

- [ ] Discuss and maybe remove original single_sc calculation
- [ ] Add ib calculation on bus for generator-close short calculation
- [ ] Further updates in test

**Comment by wangzhenassd** [comment]
> It seems like the `calc_single_sc` is not yet implemented for 1ph short-circuits. This is also something that would be nice to have.
> 
> #### `calc_sc`:
> https://github.com/e2nIEE/pandapower/blob/c9aaecba98c6a0c812452a82fb7064b82bde1c70/pandapower/shortcircuit/calc_sc.py#L128-L133
> 
> #### `calc_single_sc`:
> https://github.com/e2nIEE/pandapower/blob/c9aaecba98c6a0c812452a82fb7064b82bde1c70/pandapower/shortcircuit/calc_sc.py#L198-L201

Hi,
I think after #1082 merged, the functionality is also fulfilled.

**Comment by wangzhenassd** [comment]
@lthurner @rbolgaryn Do we want to maybe remove the original calc_single_sc function? since in PR the calc_sc can already define the required bus and also use the fast version.

**Comment by lthurner** [comment]
Yes calc_single_sc can be removed @wangzhenassd
